pkg/usagestats/reporter_test.go

```diff
@@ -0,0 +1,221 @@
+	t.Logf("data: %+v", data.(*ClusterSeed))
+	require.Equal(t, data.(*ClusterSeed).UID, first)
+}
+
+func Test_ReportLoop(t *testing.T) {
+	// stub
+	reportCheckInterval = 100 * time.Millisecond
+	reportInterval = time.Second
+	stabilityCheckInterval = 100 * time.Millisecond
+
+	totalReport := 0
+	clusterIDs := []string{}
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		var received Report
+		totalReport++
+		require.NoError(t, jsoniter.NewDecoder(r.Body).Decode(&received))
+		clusterIDs = append(clusterIDs, received.ClusterID)
+		rw.WriteHeader(http.StatusOK)
+	}))
+	usageStatsURL = server.URL
+
+	objectClient, err := local.NewBackend(&local.Config{
+		Path: t.TempDir(),
+	})
+
+	require.NoError(t, err)
+
+	r, err := NewReporter(Config{Leader: true, Enabled: true}, kv.Config{
+		Store: "inmemory",
+	}, objectClient, objectClient, log.NewLogfmtLogger(os.Stdout), prometheus.NewPedanticRegistry())
+	require.NoError(t, err)
+	ctx, cancel := context.WithCancel(context.Background())
+	r.initLeader(ctx)
+
+	go func() {
+		<-time.After(6*time.Second + (stabilityCheckInterval * time.Duration(stabilityMinimunRequired+1)))
+		cancel()
+	}()
+	require.Equal(t, context.Canceled, r.running(ctx))
+	require.GreaterOrEqual(t, totalReport, 5)
+	first := clusterIDs[0]
+	for _, uid := range clusterIDs {
+		require.Equal(t, first, uid)
+	}
+	require.Equal(t, first, r.cluster.UID)
+}
+
+func Test_NextReport(t *testing.T) {
+	fixtures := map[string]struct {
+		interval  time.Duration
+		createdAt time.Time
+		now       time.Time
+
+		next time.Time
+	}{
+		"createdAt aligned with interval and now": {
+			interval:  1 * time.Hour,
+			createdAt: time.Unix(0, time.Hour.Nanoseconds()),
+			now:       time.Unix(0, 2*time.Hour.Nanoseconds()),
+			next:      time.Unix(0, 2*time.Hour.Nanoseconds()),
+		},
+		"createdAt aligned with interval": {
+			interval:  1 * time.Hour,
+			createdAt: time.Unix(0, time.Hour.Nanoseconds()),
+			now:       time.Unix(0, 2*time.Hour.Nanoseconds()+1),
+			next:      time.Unix(0, 3*time.Hour.Nanoseconds()),
+		},
+		"createdAt not aligned": {
+			interval:  1 * time.Hour,
+			createdAt: time.Unix(0, time.Hour.Nanoseconds()+18*time.Minute.Nanoseconds()+20*time.Millisecond.Nanoseconds()),
+			now:       time.Unix(0, 2*time.Hour.Nanoseconds()+1),
+			next:      time.Unix(0, 2*time.Hour.Nanoseconds()+18*time.Minute.Nanoseconds()+20*time.Millisecond.Nanoseconds()),
+		},
+	}
+	for name, f := range fixtures {
+		t.Run(name, func(t *testing.T) {
+			next := nextReport(f.interval, f.createdAt, f.now)
+			require.Equal(t, f.next, next)
+		})
+	}
+}
+
+func TestWrongKV(t *testing.T) {
+	objectClient, err := local.NewBackend(&local.Config{
+		Path: t.TempDir(),
+	})
+	require.NoError(t, err)
+
+	r, err := NewReporter(Config{Leader: true, Enabled: true}, kv.Config{
+		Store: "",
+	}, objectClient, objectClient, log.NewLogfmtLogger(os.Stdout), prometheus.NewPedanticRegistry())
+	require.NoError(t, err)
```

Comment:
I don't think so, because only the ingesters will require the k/v store.  This will get checked when the ingester is started during the call to `running()`.